src/components/scenes/CreateWalletSelectFiatScene.tsx

```diff
@@ -1,161 +1,225 @@
+        const removedItem = newWalletItemsCopy.splice(i, 1)
+        importNotSupportedItems.push(removedItem[0])
       }
-    } else {
-      Alert.alert(s.strings.create_wallet_invalid_input, s.strings.create_wallet_select_valid_fiat)
     }
-  }
-
-  handleSearchTermChange = (searchTerm: string) => {
-    this.setState({
-      searchTerm
-    })
-  }
 
-  handleSelectFiatType = (item: GuiFiatType) => {
-    const selectedFiat = this.props.supportedFiats.find(type => type.value === item.value)
+    // Check if any remaining selected assets can be imported
+    if (newWalletItemsCopy.length === 0 && newTokenItems.length === 0) {
+      await Airship.show<'cancel' | undefined>(bridge => (
+        <ButtonsModal
+          bridge={bridge}
+          title={s.strings.create_wallet_failed_import_header}
+          message={s.strings.create_wallet_all_disabled_import}
+          buttons={{
+            cancel: { label: s.strings.string_cancel_cap }
+          }}
+        />
+      ))
+
+      return
+    }
 
-    if (selectedFiat) {
-      this.setState(
-        {
-          selectedFiat: selectedFiat.value
-        },
-        this.onNext
-      )
+    // If items remain that can be imported/enabled, show a warning that some assets will cannot be imported
+    if (importNotSupportedItems.length > 0) {
+      const displayNames = importNotSupportedItems.map(item => item.displayName).join(', ')
+      const resolveValue = await Airship.show<'continue' | 'cancel' | undefined>(bridge => (
+        <ButtonsModal
+          bridge={bridge}
+          title={s.strings.create_wallet_failed_import_header}
+          message={sprintf(s.strings.create_wallet_some_disabled_import, displayNames)}
+          buttons={{
+            continue: { label: s.strings.legacy_address_modal_continue },
+            cancel: { label: s.strings.string_cancel_cap }
+          }}
+        />
+      ))
+
+      if (resolveValue === 'cancel' || resolveValue == null) {
+        return
+      }
     }
-  }
 
-  renderFiatTypeResult = (data: FlatListItem<GuiFiatType>) => {
-    const styles = getStyles(this.props.theme)
-    const fiatCountry = FIAT_COUNTRY[data.item.value]
-    if (!fiatCountry) {
-      return null
+    // If all remaining create items are tokens just go enable them and return home
+    if (newWalletItemsCopy.length === 0 && newTokenItems.length > 0) {
+      await dispatch(enableTokensAcrossWallets(newTokenItems))
+      navigation.navigate('walletListScene', {})
+      return
     }
 
+    navigation.navigate('createWalletImport', { createWalletList: [...newWalletItemsCopy, ...newTokenItems], walletNames, fiatCode: fiat.value })
+  })
+
+  const renderSelectedFiatRow = useHandler(() => {
+    const fiatCountry = FIAT_COUNTRY[fiat.value]
+    // @ts-expect-error
+    const subTitle = s.strings[`currency_label_${fiat.value}`] ?? s.strings.currency_label_
```

Comment:
Is `s.strings.currency_label_` an error? Since you have the `// @ts-expect-error` for the first half of the statement, it might also be hiding an error in the second half.

You can usually avoid `// @ts-expect-error` by using `any`, which is preferable because it's more targeted:

```ts
const key: any = `currency_label_${fiat.value}`
const subTitle = s.strings[key] ?? s.strings.currency_label_WHAT
```

Reply:
Still getting a type error when using any. I was able to satisfy TS with the following:

```
const key = `currency_label_${fiat.value}`
const subTitle = s.strings[key as keyof typeof s.strings] ?? s.strings.currency_label_
```

Reply:
That works for me.